source/system/commands/SetEntryPropertyCommand.js

```diff
@@ -26,6 +26,12 @@ class SetEntryPropertyCommand extends BaseCommand {
             throw new Error("Entry not found for ID");
         }
         entry[propertyName] = value;
+
+        if (propertyName === "password") {
+            entry.attributes = entry.attributes || {};
+            entry.attributes["passModifiedTime"] = (new Date()).getTime();
```

Comment:
Unfortunately this won't work as is. SetEntryProperty is called every time an archive is loaded, which means the password would be marked as modified each time. A new command should be executed from within `setProperty` on `Entry`: Every time `"password"` is set, set the attribute as well.